src/renderer/components/settings-execution.tsx

```diff
@@ -12,29 +14,69 @@ import * as React from 'react';
 import { IPackageManager } from '../npm';
 import { AppState } from '../state';
 
+export enum SettingItemType {
+  EnvVar,
+  Flag,
+}
+
 interface ExecutionSettingsProps {
   appState: AppState;
 }
 
+interface ExecutionSettingsState {
+  executionFlags: Record<string, string>;
+  environmentVariables: Record<string, string>;
+}
+
 /**
  * Settings content to manage execution-related preferences.
  *
  * @class ExecutionSettings
  * @extends {React.Component<ExecutionSettingsProps, {}>}
  */
 @observer
-export class ExecutionSettings extends React.Component<ExecutionSettingsProps> {
+export class ExecutionSettings extends React.Component<
+  ExecutionSettingsProps,
+  ExecutionSettingsState
+> {
   constructor(props: ExecutionSettingsProps) {
     super(props);
 
+    this.state = {
+      executionFlags: Object.assign(
+        {},
+        ...props.appState.executionFlags.map((flag, idx) => {
+          return { [`flag-${idx}`]: flag };
+        }),
+      ),
```

Comment:
Is Object.assign needed?

```suggestion
      executionFlags: {
        ...props.appState.executionFlags.map((flag, idx) => {
          return { [`flag-${idx}`]: flag };
        }),
      },
```

Reply:
yes - those code blocks aren't equivalent :)

<img width="667" alt="Screen Shot 2021-05-01 at 10 50 22 AM" src="https://user-images.githubusercontent.com/2036040/116776848-20983280-aa6b-11eb-835d-4e9e67198f4a.png">